CMakeLists.txt

```diff
@@ -0,0 +1,160 @@
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+cmake_minimum_required(VERSION 3.2 FATAL_ERROR)
+project(date VERSION 2.1 LANGUAGES CXX)
+
+if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_CURRENT_SOURCE_DIR})
+	set(NEED_PARENT Off)
+	set(DATE_DEFAULT_ENABLE_TESTS On)
+else()
+	set(NEED_PARENT On)
+	set(DATE_DEFAULT_ENABLE_TESTS Off)
+endif()
+
+find_package(CURL)
+
+if(WIN32)
+	find_program(DATE_TZ_SHELL_REMOVE_DIRECTORY "rd")
+	find_program(DATE_TZ_SHELL_REMOVE_FILE "del")
+	find_program(DATE_TZ_SHELL_MOVE_FILE "move")
+	find_program(DATE_TZ_SHELL_UNTAR "7z.exe" PATHS "$ENV{ProgramFiles}/7-Zip")
+else()
+	find_program(DATE_TZ_SHELL_REMOVE_DIRECTORY "rm")
+	find_program(DATE_TZ_SHELL_REMOVE_FILE "rm")
+	find_program(DATE_TZ_SHELL_MOVE_FILE "mv")
+	find_program(DATE_TZ_SHELL_UNTAR "tar")
+endif()
+find_program(DATE_TZ_SHELL_MAKE_DIRECTORY "mkdir")
+
+option(DATE_ENABLE_TESTS "Build unit tests" ${DATE_DEFAULT_ENABLE_TESTS})
+option(DATE_NO_TZ        "Disable building Time Zone static library" Off)
+if(NOT DATE_NO_TZ)
+	option(DATE_TZ_NO_LAZY   "Disable lazy initialization of time zones" Off)
+	if(CURL_FOUND)
+		message("Found cURL, enabling remote")
+		option(DATE_TZ_NO_REMOTE "Disable using cURL to download IANA Time Zone rules" Off)
+	else()
+		message(WARNING "Could not find cURL libraries. Disabling remote.")
+		option(DATE_TZ_NO_REMOTE "Disable using cURL to download IANA Time Zone rules" On)
+	endif()
+	if(WIN32)
+		option(DATE_TZ_NO_TZMAP  "Disable Time Zone name mapping on Windows" Off)
+	else()
+		option(DATE_TZ_NO_TZMAP  "Disable Time Zone name mapping on Windows" On)
+	endif()
+	if(DATE_TZ_SHELL_REMOVE_DIRECTORY AND DATE_TZ_SHELL_REMOVE_FILE AND DATE_TZ_SHELL_MOVE_FILE AND DATE_TZ_SHELL_UNTAR AND DATE_TZ_SHELL_MAKE_DIRECTORY)
+		option(DATE_TZ_NO_DATE_TZ_SHELL  "Disable using shell programs to manage IANA Time Zone rules" Off)
+	else()
+		option(DATE_TZ_NO_DATE_TZ_SHELL  "Disable using shell programs to manage IANA Time Zone rules" On)
+	endif()
+endif()
+
+message(STATUS "Using ${CMAKE_CXX_COMPILER} (compiler id: ${CMAKE_CXX_COMPILER_ID})")
+if ("${CMAKE_CXX_COMPILER_ID}" MATCHES ".*Clang")
+	if(NOT "${CMAKE_CXX_FLAGS}" MATCHES "-stdlib")
+		set(CMAKE_CXX_FLAGS "-std=c++14 -stdlib=libc++ ${CMAKE_CXX_FLAGS}")
```

Comment:
This assumes that the user has libc++, which is only true if they built it from source on most systems.